### PR TITLE
Alarms py collector add filtering

### PR DESCRIPTION
**Pull request by andrewm4894**
##### Summary

Adds ability to filter the alarms collected by the python alarms collector.

##### Test Plan

Tested locally with various filters applied.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

Ability to just collect a subset of alarms with the alarms collector.

### CODE REVIEW

File: collectors/python.d.plugin/alarms/alarms.chart.py [hunk 1]
```diff
@@ -57,6 +58,9 @@ def _get_data(self):
 
         raw_data = loads(raw_data)
         alarms = raw_data.get('alarms', {})
+        if self.alarm_contains_words != '':
+            alarms = {alarm_name: alarms[alarm_name] for alarm_name in alarms for alarm_contains_word in
+                      self.alarm_contains_words.split(',') if alarm_contains_word in alarm_name}
```
Comment (ilyam8) [inline collectors/python.d.plugin/alarms/alarms.chart.py:63]: Perhaps to make a set in init() and just "x in y" here?

Comment (ilyam8) [reply collectors/python.d.plugin/alarms/alarms.chart.py:63]: More short version (requires split self.alarm_contains_words in init()):

```py
{name: value for name, value in alarms.items() if any(w in name for w in self.alarm_contains_words)}
```

File: collectors/python.d.plugin/alarms/alarms.chart.py [hunk 2]
```diff
@@ -49,6 +49,7 @@ def __init__(self, configuration=None, name=None):
         self.url = self.configuration.get('url', DEFAULT_URL)
         self.collect_alarm_values = bool(self.configuration.get('collect_alarm_values', DEFAULT_COLLECT_ALARM_VALUES))
         self.collected_dims = {'alarms': set(), 'values': set()}
+        self.alarm_contains_words = self.configuration.get('alarm_contains_words', DEFAULT_ALARM_CONTAINS_WORDS)
```
Comment (ilyam8) [inline collectors/python.d.plugin/alarms/alarms.chart.py:52]: I suggest splitting and, optionally, trimming spaces (e.g. hadnle `cpu, load`) here

**Review by DShreve2 — APPROVED** [review]

File: collectors/python.d.plugin/alarms/alarms.chart.py (hunk 2 above)
Comment (andrewm4894) [reply collectors/python.d.plugin/alarms/alarms.chart.py:52]: Will do

File: collectors/python.d.plugin/alarms/alarms.chart.py (hunk 1 above)
Comment (andrewm4894) [reply collectors/python.d.plugin/alarms/alarms.chart.py:63]: Will do

Comment (andrewm4894) [reply collectors/python.d.plugin/alarms/alarms.chart.py:63]: i actually tried this first, doing the split() in __init__ and was not working for some reason.

Comment (andrewm4894) [reply collectors/python.d.plugin/alarms/alarms.chart.py:63]: will have another go as would be better to do it all in init

**Review by ilyam8 — APPROVED** [review]

File: collectors/python.d.plugin/alarms/alarms.chart.py (hunk 2 above)
Comment (andrewm4894) [reply collectors/python.d.plugin/alarms/alarms.chart.py:52]: done

**Comment by andrewm4894** [comment]
@DShreve2 can i get another approve here please (my last commit i think made your first approve stale).

**Review by vlvkobal — APPROVED** [review]

**Pull request merged**